packages/ember-data/tests/integration/setup-container-test.js

```diff
@@ -63,7 +67,7 @@ test("the deprecated adapter:_rest is resolved as adapter:rest", function() {
     deprecated = container.lookup('adapter:_rest');
   });
 
-  ok(deprecated === valid, "they should resolve to the same thing");
+  ok(deprecated._debugContainerKey === valid._debugContainerKey, "they should resolve to the same thing");
```

Comment:
comparing keys like this is just wrong, I know. `deprecated` and `valid` are classes so we likely should use `deepEqual`